src/psyclone/psyir/transformations/hoist_local_arrays_trans.py

```diff
@@ -350,9 +347,19 @@ def _get_local_arrays(node):
             # statement, not the hoisted declaration
             if isinstance(sym.datatype.intrinsic, DataTypeSymbol):
```

Comment:
Here, it would be better to use `sym.reference_accesses()`, just in case e.g. the precision corresponds to an expression (e.g. I think it's valid to have `2*wp`). This would also avoid having to consider the precision and the intrinsic type separately. Should the comment at L344 say "which are not global" rather than "local"?